tests/test/unify.rs

```diff
@@ -333,3 +333,19 @@ fn quantified_types() {
         }
     }
 }
+
+#[test]
+fn subtype() {
```

Comment:
I honestly have no clue about tests.

Reply:
Well, you might test the region constraints that arise from `&'a u32 <: &'b u32`, for example -- but I can push some more tests later on. If we had the ability to annotate structs with variance etc I guess that'd be better.